apps/ftu/js/memcard_manager.js

```diff
@@ -31,81 +35,103 @@ var SdManager = {
 
     var cancelled = false;
     var importer = null;
+    var importButton;
 
     var progress = utils.overlay.show(
-      'memoryCardContacts-reading', 'activityBar');
+      'memoryCardContacts-importing',
+      'activityBar',
+      'infiniteProgress'
+    );
     utils.overlay.showMenu();
-    utils.overlay.oncancel = function() {
-      cancelled = true;
-      if (importer) {
-        importer.finish();
-      } else {
-        UIManager.navBar.removeAttribute('aria-disabled');
-        utils.overlay.hide();
-      }
-    };
 
-    var importButton = UIManager.sdImportButton;
+    return new Promise((resolve, reject) => {
+      utils.overlay.oncancel = () => {
+        cancelled = true;
+        if (importer) {
+          importer.finish();
+        } else {
+          UIManager.navBar.removeAttribute('aria-disabled');
+          utils.overlay.hide();
+        }
+        resolve();
+      };
 
+      importButton = UIManager.sdImportButton;
 
-    utils.sdcard.retrieveFiles([
-      'text/vcard',
-      'text/directory;profile=vCard',
-      'text/directory'
-    ], ['vcf', 'vcard'], function(err, fileArray) {
-      if (err) {
-        return import_error(err);
-      }
-      if (cancelled) {
-        return;
-      }
-      if (fileArray.length) {
-        utils.sdcard.getTextFromFiles(fileArray, '', onFiles);
-      } else {
-        import_error('No contacts were found.');
-      }
-    });
+      utils.sdcard.retrieveFiles([
+        'text/vcard',
+        'text/directory;profile=vCard',
+        'text/directory'
+      ], ['vcf', 'vcard'], (error, fileArray) => {
+        if (!error && !fileArray.length) {
+          error = 'No contacts were found';
+        }
 
-    function onFiles(err, text) {
-      if (err) {
-        return import_error(err);
-      }
-      if (cancelled) {
-        return;
-      }
-      importer = new VCFReader(text);
-      if (!text || !importer) {
-        return import_error('No contacts were found.');
-      }
+        if (error) {
+          reject(error);
+          return import_error(error);
+        }
 
-      importer.onread = import_read;
-      importer.onimported = imported_contact;
-      importer.onerror = import_error;
+        if (cancelled) {
+          resolve();
+          return;
+        }
 
-      importer.process(function import_finish() {
-        window.setTimeout(function onfinish_import() {
-          utils.misc.setTimestamp('sd');
-          UIManager.navBar.removeAttribute('aria-disabled');
-          utils.overlay.hide();
-          if (!cancelled) {
+        var promises = [];
+        fileArray.forEach(file => {
+          promises.push(utils.sdcard.getTextFromFile(file, onContacts));
+        });
+
+        Promise.all(promises).then(() => {
+          window.setTimeout(() => {
```

Comment:
I don't think we will need this `window.setTimeout` since we have the promise that is already async